lib/svg-sprite/transform/svgo.js

```diff
@@ -70,15 +30,20 @@ var SVGO = require('svgo'),
  */
 module.exports = function (shape, config, spriter, cb) {
     config = _.cloneDeep(config);
-    config.plugins = ('plugins' in config) ? defaultPluginConfig.concat(config.plugins) : defaultPluginConfig;
-    config.plugins.push({ removeXMLProcInst: !!spriter.config.svg.xmlDeclaration });
-    config.plugins.push({ removeDoctype: !!spriter.config.svg.doctypeDeclaration });
+    config.plugins = svgo.extendDefaultPlugins('plugins' in config ? defaultPluginConfig.concat(config.plugins) : defaultPluginConfig);
+    config.plugins.push({
+        name: 'removeXMLProcInst',
+        active: !!spriter.config.svg.xmlDeclaration
+    }, {
+        name: 'removeDoctype',
+        active: !!spriter.config.svg.doctypeDeclaration
+    });
 
     var svg = shape.getSVG(false),
-        svgLength = svg.length,
-        svgoInstance = new SVGO(config);
+        svgLength = svg.length;
 
-    svgoInstance.optimize(svg).then(function (result) {
+    try {
+        var result = svgo.optimize(svg, config);
```

Comment:
Need another pair of eyes but this should be synchronous in v2.x.

Reply:
Yeah, you should be right! Looks good to me.